### ISSUE TITLE
Support for a new Z-wave device

### ISSUE THREAD

**Issue by xsnrg** [cdjackson/HABmin2#38]
Chris, what is your preferred method of requesting a new device to be added?  If this ticket suffices, here is a link to the device.  If there is anything else you need, or I missed a wiki link on how to get you everything you need, please direct appropriately.

http://products.z-wavealliance.org/products/1179

I was able to bind the device, and it shows up correctly as an unknown device.

Thanks,
Jim

**Comment by cdjackson** [comment]
Hi Jim,
Here is fine. But I need a bit more info (and you're right that there might not be somewhere this is written down, so I'll sort that out as well!).

I need to know the type and ID that HABmin prints when it doesn't know the name. I also need to know the configuration. This can come from the manual, or ideally from the pepper1 database (http://www.pepper1.net/zwavedb/) - either way, at least one of these is needed.

In this case, I suspect it will be very similar to the old sensor, but this is the Gen5 version (I think) so I can start with that, but if you can provide the link to the pepper database, that would be good (if it's in there - if not, let me know but I will definately need the type/id)

**Comment by xsnrg** [comment]
Thanks Chris, let me see what I can find.  Yes, this is the Gen5 version.

Habmin2 is printing:

Manufacturer: Aeon Labs
Manufacturer ID: 86
Device ID: 59
Device Type: 102
Version: 4
Protocol version 3.92
Application version 1.13

I looked in pepper1, it does not seem to be there yet.  I messed around trying to get data from it into habmin, and got the above data, as well as battery reporting in correctly.  I will keep digging to see what else I can find that may be helpful.  The command classes are listed in the link on the opening post when you click on "view", though not sure that helps much.

Jim

**Comment by xsnrg** [comment]
Pulled these out of the logs as well.  I would guess some of these are Gen5 related:

2015-05-10 18:49:51.883 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass] - NODE 3: Unsupported command class ZWAVE_PLUS_INFO
2015-05-10 18:49:51.884 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass] - NODE 3: Unsupported command class ASSOCIATION_GROUP_INFO
2015-05-10 18:49:51.885 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass] - NODE 3: Unsupported command class POWERLEVEL
2015-05-10 18:49:51.886 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass] - NODE 3: Unsupported command class FIRMWARE_UPDATE_MD
2015-05-10 18:49:51.886 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass] - NODE 3: Unsupported command class SECURITY

**Comment by cdjackson** [comment]
Thanks Jim. I'll try and look at this tonight if I can (although I'm on a plane for a few hours :( ).  I'll probably copy the old version of this as a starter (I'm guessing it's pretty similar as the old one is still quite new).

Don't worry about the unsupported command class warnings - this shouldn't matter...

**Comment by xsnrg** [comment]
I reached out to Aeon Labs and asked them to help out by updating pepper1 if they can.  I really like their products, hopefully their community support will be as good.

**Comment by cdjackson** [comment]
Ok - lets give it a few days. I've found their support to be quite good...

**Comment by xsnrg** [comment]
Dropped you an email with an attachment from Aeotec.  Hope it has what we need.  Looks like I have a good contact there now for other things as well.

**Comment by cdjackson** [comment]
Yep - got it - thanks. It’s the info I was after…

**Comment by cdjackson** [comment]
This is included - see the link (https://github.com/openhab/openhab/pull/2601) where you'll find a link to cloudbees. I'll merge this today and it will be in the snapshot tomorrow, but if you want to try before then, use the link above...

**Comment by xsnrg** [comment]
Chris, it looks great, thanks for throwing it in there.  I noticed an extra character to clean up in the device:

**Comment by cdjackson** [comment]
Great - Did something get trimmed off your message? I feel there should be a picture or something? :)

**Comment by xsnrg** [comment]
That it did.. wow you are fast :)

**Comment by xsnrg** [comment]
A happy side effect, without changing anything else, a rule that had not been working for a zwave motion sensor started working after dropping the new zwave jar in.

**Comment by xsnrg** [comment]
It appears it was pulling information previously discovered and put in userdata/zwave/node3.xml.  After removing this file and restarting OH a few times, the file for this node is not getting re-created, even after poking the wake-up button.  A lot of data hits the logs when it is woken up, but no xml, and events seem to not be created.
Some other strange things:

The device class says SIMPLE_GARAGE_DOOR

Output from manual wake-up by button press (lines containing NODE 3)

2015-05-16 17:37:07 [DEBUG] [ApplicationCommandMessageClass:38   ] - NODE 3: Application Command Request (ALIVE:GET_CONFIGURATION)
2015-05-16 17:37:07 [DEBUG] [ApplicationCommandMessageClass:56   ] - NODE 3: Incoming command class BASIC
2015-05-16 17:37:07 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:74   ] - NODE 3: Received Basic Request
2015-05-16 17:37:07 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:78   ] - NODE 3: Basic Set sent to the controller will be processed as Basic Report
2015-05-16 17:37:07 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:107  ] - NODE 3: Basic report, value = 0xFF
2015-05-16 17:37:07 [DEBUG] [.z.internal.ZWaveActiveBinding:450  ] - NODE 3: Got a value event from Z-Wave network, endpoint = 0, command class = BASIC, value = 255
2015-05-16 17:37:13 [DEBUG] [.ApplicationUpdateMessageClass:44   ] - NODE 3: Application update request. Node information received.
2015-05-16 17:37:13 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass:221  ] - NODE 3: Unsupported command class ZWAVE_PLUS_INFO
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class SENSOR_BINARY
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class BATTERY
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class WAKE_UP
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class CONFIGURATION
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class ASSOCIATION
2015-05-16 17:37:13 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass:221  ] - NODE 3: Unsupported command class ASSOCIATION_GROUP_INFO
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class ALARM
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class VERSION
2015-05-16 17:37:13 [DEBUG] [.o.b.z.i.p.c.ZWaveCommandClass:224  ] - NODE 3: Creating new instance of command class MANUFACTURER_SPECIFIC
2015-05-16 17:37:13 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass:221  ] - NODE 3: Unsupported command class POWERLEVEL
2015-05-16 17:37:13 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass:221  ] - NODE 3: Unsupported command class FIRMWARE_UPDATE_MD
2015-05-16 17:37:13 [WARN ] [.o.b.z.i.p.c.ZWaveCommandClass:221  ] - NODE 3: Unsupported command class SECURITY
2015-05-16 17:37:13 [DEBUG] [.i.p.c.ZWaveWakeUpCommandClass:411  ] - NODE 3: Is awake with 2 messages in the wake-up queue.
2015-05-16 17:37:13 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:1028 ] - NODE 3: Wakeup during initialisation.
2015-05-16 17:37:13 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:267  ] - NODE 3: Node advancer - GET_CONFIGURATION: queue length(1), free to send(false)
2015-05-16 17:37:13 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:333  ] - NODE 3: Node advancer: loop - GET_CONFIGURATION try 3: stageAdvanced(false)
2015-05-16 17:37:13 [DEBUG] [ZWaveConfigurationCommandClass:154  ] - NODE 3: Creating new message for application command CONFIGURATIONCMD_GET
2015-05-16 17:37:13 [DEBUG] [o.b.z.i.protocol.SerialMessage:109  ] - NODE 3: Creating empty message of class = SendData (0x13), type = Request (0x00)
2015-05-16 17:37:13 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:240  ] - NODE 3: Node advancer - queued packet. Queue length is 1
2015-05-16 17:37:13 [DEBUG] [WaveController$ZWaveSendThread:1315 ] - NODE 3: Sending REQUEST Message = 01 0A 00 13 03 03 70 05 FF 25 3E 77 
2015-05-16 17:37:13 [DEBUG] [b.z.i.p.s.SendDataMessageClass:38   ] - NODE 3: Sent Data successfully placed on stack.
2015-05-16 17:37:13 [DEBUG] [b.z.i.p.s.SendDataMessageClass:74   ] - NODE 3: SendData Request. CallBack ID = 62, Status = Transmission complete and ACK received(0)
2015-05-16 17:37:16 [DEBUG] [ApplicationCommandMessageClass:38   ] - NODE 3: Application Command Request (ALIVE:GET_CONFIGURATION)
2015-05-16 17:37:16 [DEBUG] [ApplicationCommandMessageClass:56   ] - NODE 3: Incoming command class BASIC
2015-05-16 17:37:16 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:74   ] - NODE 3: Received Basic Request
2015-05-16 17:37:16 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:78   ] - NODE 3: Basic Set sent to the controller will be processed as Basic Report
2015-05-16 17:37:16 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:107  ] - NODE 3: Basic report, value = 0x00
2015-05-16 17:37:16 [DEBUG] [.z.internal.ZWaveActiveBinding:450  ] - NODE 3: Got a value event from Z-Wave network, endpoint = 0, command class = BASIC, value = 0
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:1005 ] - NODE 3: Node advancer - GET_CONFIGURATION: Transaction complete (SendData:Request) success(true)
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:201  ] - NODE 3: Node advancer - checking initialisation queue. Queue size 1.
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:207  ] - NODE 3: Node advancer - message removed from queue. Queue size 0.
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:267  ] - NODE 3: Node advancer - GET_CONFIGURATION: queue length(0), free to send(true)
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:333  ] - NODE 3: Node advancer: loop - GET_CONFIGURATION try 4: stageAdvanced(false)
2015-05-16 17:37:16 [DEBUG] [ZWaveConfigurationCommandClass:154  ] - NODE 3: Creating new message for application command CONFIGURATIONCMD_GET
2015-05-16 17:37:16 [DEBUG] [o.b.z.i.protocol.SerialMessage:109  ] - NODE 3: Creating empty message of class = SendData (0x13), type = Request (0x00)
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:240  ] - NODE 3: Node advancer - queued packet. Queue length is 1
2015-05-16 17:37:16 [DEBUG] [WaveController$ZWaveSendThread:1374 ] - NODE 3: Response processed after 3624ms/3951ms.
2015-05-16 17:37:16 [DEBUG] [WaveController$ZWaveSendThread:1315 ] - NODE 3: Sending REQUEST Message = 01 09 00 13 03 02 84 08 25 3F 72 
2015-05-16 17:37:16 [DEBUG] [b.z.i.p.s.SendDataMessageClass:38   ] - NODE 3: Sent Data successfully placed on stack.
2015-05-16 17:37:16 [DEBUG] [b.z.i.p.s.SendDataMessageClass:74   ] - NODE 3: SendData Request. CallBack ID = 63, Status = Transmission complete and ACK received(0)
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:1005 ] - NODE 3: Node advancer - GET_CONFIGURATION: Transaction complete (SendData:Request) success(true)
2015-05-16 17:37:16 [DEBUG] [z.i.p.i.ZWaveNodeStageAdvancer:201  ] - NODE 3: Node advancer - checking initialisation queue. Queue size 1.
2015-05-16 17:37:16 [DEBUG] [.i.p.c.ZWaveWakeUpCommandClass:372  ] - NODE 3: Went to sleep
2015-05-16 17:37:16 [DEBUG] [.i.p.c.ZWaveWakeUpCommandClass:435  ] - NODE 3: Is sleeping
2015-05-16 17:37:16 [DEBUG] [WaveController$ZWaveSendThread:1374 ] - NODE 3: Response processed after 25ms/3951ms.
2015-05-16 17:37:16 [DEBUG] [.i.p.c.ZWaveWakeUpCommandClass:241  ] - NODE 3: Putting message SendData in wakeup queue.
2015-05-16 17:37:16 [DEBUG] [.i.p.c.ZWaveWakeUpCommandClass:237  ] - NODE 3: Message already on the wake-up queue. Removing original.
2015-05-16 17:37:16 [DEBUG] [.i.p.c.ZWaveWakeUpCommandClass:241  ] - NODE 3: Putting message SendData in wakeup queue.


Lines returned from a open/close event (which seems to work okay):


2015-05-16 17:38:00 [DEBUG] [ApplicationCommandMessageClass:38   ] - NODE 3: Application Command Request (ALIVE:GET_CONFIGURATION)
2015-05-16 17:38:00 [DEBUG] [ApplicationCommandMessageClass:56   ] - NODE 3: Incoming command class BASIC
2015-05-16 17:38:00 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:74   ] - NODE 3: Received Basic Request
2015-05-16 17:38:00 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:78   ] - NODE 3: Basic Set sent to the controller will be processed as Basic Report
2015-05-16 17:38:00 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:107  ] - NODE 3: Basic report, value = 0xFF
2015-05-16 17:38:00 [DEBUG] [.z.internal.ZWaveActiveBinding:450  ] - NODE 3: Got a value event from Z-Wave network, endpoint = 0, command class = BASIC, value = 255
2015-05-16 17:38:03 [DEBUG] [ApplicationCommandMessageClass:38   ] - NODE 3: Application Command Request (ALIVE:GET_CONFIGURATION)
2015-05-16 17:38:03 [DEBUG] [ApplicationCommandMessageClass:56   ] - NODE 3: Incoming command class BASIC
2015-05-16 17:38:03 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:74   ] - NODE 3: Received Basic Request
2015-05-16 17:38:03 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:78   ] - NODE 3: Basic Set sent to the controller will be processed as Basic Report
2015-05-16 17:38:03 [DEBUG] [z.i.p.c.ZWaveBasicCommandClass:107  ] - NODE 3: Basic report, value = 0x00
2015-05-16 17:38:03 [DEBUG] [.z.internal.ZWaveActiveBinding:450  ] - NODE 3: Got a value event from Z-Wave network, endpoint = 0, command class = BASIC, value = 0

**Comment by xsnrg** [comment]
btw, if the previous is just a bunch of junk, would like to be more helpful.  Just point me in the right direction.

Thanks
Jim

**Comment by xsnrg** [comment]
One last comment, the battery is not reporting in correctly after removing the xml file either.

Jim

**Comment by cdjackson** [comment]
It will take some time for the XML to be generated again - it needs to complete initialisation. It’s hard to tell what’s happening as you’ve grepped the log and I’d really need to see the other lines. It might be stuck in the get configuration mode, or it might just have not completed yet.

**Comment by xsnrg** [comment]
Correct you are.  This morning it had been generated.  When I went to bed it was still in get configuration mode.  I just figured since the other 3 devices were generated right away this might be an anomaly so wanted to let you know.  The battery is also reporting in now.  I wonder why it did not get what it wanted from the forced wakeup...

It looks like the only strange things left then are the extra parenthesis in the Product name, and that it is calling itself a garage door.

node3.xml:    <genericDeviceClass>GARAGE_DOOR</genericDeviceClass>
node3.xml:    <specificDeviceClass>SIMPLE_GARAGE_DOOR</specificDeviceClass>

The irony of course is that I chose to put the device in my door to the garage :)

**Comment by cdjackson** [comment]
Don’t worry about the garage door thing - this is sort of my fault… By this I mean when I first added this device class, something I read indicated it was for a garage door, so that’s what it’s named. I’ll probably change this to something more generic, but it’s not used for anything so don’t worry about it…

There may be a problem with initialisation - that might be why it took so long. If you have the log, feel free to send it to me… If the device doesn’t respond to some requests, then it will take a long time to initialise - if I know this, I can add info to the database to stop it making these requests...

**Comment by xsnrg** [comment]
Thanks Chris.  I added the config to logback, removed all the log and xml files, and hopefully have a good story for you now with associated zwave-only logs.  It is reproducible so far, so hopefully that makes it easier to find.

I went through each node in order and forced a wakeup.  Node 1 is the stick, and it created it's file right away.  Node 2 is a motion detector, it's xml file was created when with wakeup.  Node 3 is the new gen5 door/window sensor.  It updated a lot of data with the manual wakeup, but is still not fully configured 15 minutes later, and no xml file exists yet.  Node 4 is another motion sensor that also generated the xml with wakeup.

I zipped the log file and mailed it to you to not clutter up the thread.
